clickhouse_driver/connection.py

```diff
@@ -147,25 +155,36 @@ def _create_socket(self):
             ssl_options = self.ssl_options.copy()
             ssl_options['cert_reqs'] = cert_reqs
 
-        host, port = self.host, self.port
         err = None
-        for res in socket.getaddrinfo(host, port, 0, socket.SOCK_STREAM):
-            af, socktype, proto, canonname, sa = res
-            sock = None
-            try:
-                sock = socket.socket(af, socktype, proto)
-                sock.settimeout(self.connect_timeout)
+        for host_port in self.hosts:
+            host, port = host_port.split(':')
```

Comment:
What about port [defaults](https://github.com/mymarilyn/clickhouse-driver/blob/master/clickhouse_driver/connection.py#L69)? This can be also useful.